src/main/java/net/sf/jabref/importer/fileformat/BibTeXMLImporter.java

```diff
@@ -78,33 +97,216 @@ public ParserResult importDatabase(BufferedReader reader) throws IOException {
+     * "abstract" will be put as key to fields and the value of <Code>getAbstract</Code> will be put as value to fields.
+     * Some <Code>get</Code> methods shouldn't be mapped to fields, so <Code>getClass</Code> for example will be skipped.
+     *
+     * @param t
+     * @param fields
+     */
+    private <T> void parse(T t, Map<String, String> fields) {
+        Method[] declaredMethods = t.getClass().getDeclaredMethods();
+        for (Method method : declaredMethods) {
+            try {
+                if (method.getName().equals("getYear")) {
+                    putYear(fields, (XMLGregorianCalendar) method.invoke(t));
+                    continue;
+                } else if (method.getName().equals("getNumber")) {
+                    putNumber(fields, (BigInteger) method.invoke(t));
+                    continue;
+                } else if (isFieldToSkip(method.getName())) {
+                    continue;
+                } else if (method.getName().contains("get")) {
+                    putIfValueNotNull(fields, method.getName().replace("get", ""), (String) method.invoke(t));
+                }
+            } catch (IllegalArgumentException | InvocationTargetException | IllegalAccessException e) {
+                LOGGER.error("Could not invoke method", e);
+            }
+        }
+    }
+
+    /**
+     * Returns whether the value of the given method name should be mapped or whether the field can be skipped.
+     *
+     * @param name of a method
+     * @return true if the field can be skipped, else false
+     */
+    private boolean isFieldToSkip(String name) {
+        return fieldsToSkip.contains(name);
+    }
+
+    private void parseInbook(Inbook inbook, Map<String, String> fields) {
+        List<JAXBElement<?>> content = inbook.getContent();
+        for (JAXBElement<?> element : content) {
+            String localName = element.getName().getLocalPart();
+            Object elementValue = element.getValue();
+            if (elementValue instanceof String) {
+                String value = (String) elementValue;
+                putStandardFields(localName, value, fields);
+            } else if (elementValue instanceof BigInteger) {
+                BigInteger value = (BigInteger) elementValue;
+                if (FieldName.NUMBER.equals(localName)) {
+                    putNumber(fields, value);
+                }
+                if (FieldName.CHAPTER.equals(localName) && (value != null)) {
+                    fields.put(FieldName.CHAPTER, String.valueOf(value));
+                }
+            } else if (elementValue instanceof XMLGregorianCalendar) {
+                XMLGregorianCalendar value = (XMLGregorianCalendar) elementValue;
+                if (FieldName.YEAR.equals(localName)) {
+                    putYear(fields, value);
+                }
+            }
+        }
+    }
+
+    private void putStandardFields(String localName, String value, Map<String, String> fields) {
```

Comment:
Add a comment explaining the method and where you took the "standard" fields from; e.g. from some group definition of the xsd file?